client/post-editor/editor-term-selector/index.jsx

```diff
@@ -26,21 +30,34 @@ class EditorTermSelector extends Component {
 	}
 
 	onTermsChange( selectedTerm ) {
-		const { postTerms, taxonomyName } = this.props;
+		const { postTerms, taxonomyName, siteId, postId } = this.props;
 		const terms = cloneDeep( postTerms ) || {};
-		const taxonomyTerms = terms[ taxonomyName ] || {};
+		let taxonomyTerms = map( terms[ taxonomyName ] );
+		const termIds = map( taxonomyTerms, 'ID' );
+
+		if ( includes( termIds, selectedTerm.ID ) ) {
+			taxonomyTerms = taxonomyTerms.reduce( ( prev, term ) => {
+				if ( term.ID === selectedTerm.ID ) {
+					return prev;
+				}
 
-		if ( taxonomyTerms[ selectedTerm.name ] ) {
-			delete taxonomyTerms[ selectedTerm.name ];
+				prev.push( term );
+				return prev;
+			}, [] );
 		} else {
-			taxonomyTerms[ selectedTerm.name ] = selectedTerm;
+			taxonomyTerms.push( selectedTerm );
 		}
-		terms[ taxonomyName ] = taxonomyTerms;
 
-		// TODO: REDUX - remove flux actions when whole post-editor is reduxified
-		PostActions.edit( {
-			terms: terms
-		} );
+		const taxonomyTermIds = map( taxonomyTerms, 'ID' );
+
+		this.props.editPost( {
+			terms: {
+				[ taxonomyName ]: taxonomyTerms
+			},
+			terms_by_id: {
+				[ taxonomyName ]: taxonomyTermIds.length ? taxonomyTermIds : null
+			}
+		}, siteId, postId );
```

Comment:
Hmm, why do we need to assign both `terms` and `terms_by_id` when calling `editPost` ? So that they're kept in sync in the `getEditedPost` return value? Not sure I have a great alternative, but this seems to be quite the burden in using. I'm also curious how the REST API handles this case where we'd likely be passing both `terms` and `terms_by_id` in the request.